src/mavlink_stream.c

```diff
@@ -96,16 +102,16 @@ void mavlink_stream(uint32_t time_us)
   {
     if (mavlink_streams[i].period_us && time_us - mavlink_streams[i].last_time_us >= mavlink_streams[i].period_us)
     {
-      mavlink_streams[i].last_time_us = time_us;
+      // if we took too long, set the last_time_us to be where it should have been
+      mavlink_streams[i].last_time_us = time_us - (time_us - mavlink_streams[i].last_time_us-mavlink_streams[i].period_us);
```

Comment:
I see; I think that makes sense. However, should we just simplify this to
```
mavlink_streams[i].last_time_us += mavlink_streams[i].period_us;
```
?

Also, is it possible that doing this would cause a too-fast stream request to saturate the bandwidth (we'll always take too long, so it'll always be trying to send)? Although that may happen the old way too. I guess we can try it this way for now and I'll look into it when I finally dig into the mavlink timing stuff.